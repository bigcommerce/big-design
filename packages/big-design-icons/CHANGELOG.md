# Change Log

## 1.2.2

### Patch Changes

- 98fbd46: feat(component): Add an action prop for columns whose type is number or text

## 1.2.1

### Patch Changes

- 69e7772: Remove Pin icon hardcoded fill colour

## 1.2.0

### Minor Changes

- 0f8cf34: Added Pin icon

## 1.1.0

### Minor Changes

- 01a5b20: Added AutoAwesome icon from Material Icons library

## 1.0.3

### Patch Changes

- Updated dependencies [f8665f3]
  - @bigcommerce/big-design-theme@1.1.0

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@1.0.1...@bigcommerce/big-design-icons@1.0.2) (2024-08-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [1.0.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@1.0.0...@bigcommerce/big-design-icons@1.0.1) (2024-06-11)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [1.0.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.26.1...@bigcommerce/big-design-icons@1.0.0) (2024-04-17)

### Features

- export types without reassignment ([9bf2244](https://github.com/bigcommerce/big-design/commit/9bf224417433e54680794f97d30a01fb77b24d69))

## [0.26.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.26.0...@bigcommerce/big-design-icons@0.26.1) (2024-04-09)

### Bug Fixes

- typescript build output with wrong imports ([a812497](https://github.com/bigcommerce/big-design/commit/a81249798e20c10754c938b2b3d3dcc95d81aa0c))

# [0.26.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.25.1...@bigcommerce/big-design-icons@0.26.0) (2024-04-02)

### chore

- **deps-dev:** bump flag-icons from 6.11.0 to 7.1.0 ([#1345](https://github.com/bigcommerce/big-design/issues/1345)) ([54572b6](https://github.com/bigcommerce/big-design/commit/54572b64e54cb5bb33d28c41a5cfe74844040d40))

### BREAKING CHANGES

- **deps-dev:** This upgrade removes a few flag icons. See the commit diff to check which
  ones were removed and ensure you are not using them in your application code.

## [0.25.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.25.0...@bigcommerce/big-design-icons@0.25.1) (2024-01-23)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.25.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.24.0...@bigcommerce/big-design-icons@0.25.0) (2023-10-09)

### Bug Fixes

- **icons:** downgrade 'camelcase' to 6.3.0 ([#1284](https://github.com/bigcommerce/big-design/issues/1284)) ([1004097](https://github.com/bigcommerce/big-design/commit/100409789ab0beb9095eea3580aed7b5b3fc8776))

### Features

- **icons:** add Draft icon ([#1302](https://github.com/bigcommerce/big-design/issues/1302)) ([3ae2423](https://github.com/bigcommerce/big-design/commit/3ae2423ccaf2b53f282790218dcb80d622954241))
- **icons:** add FileDownloadIcon ([#1286](https://github.com/bigcommerce/big-design/issues/1286)) ([7d73767](https://github.com/bigcommerce/big-design/commit/7d737670533c45ab09a66556b1d4b2316331ff03))
- **icons:** add window expand and minimize icons ([997400b](https://github.com/bigcommerce/big-design/commit/997400b4ba5dc8d4c484cb690ee335a112de6c32))

# [0.24.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.2...@bigcommerce/big-design-icons@0.24.0) (2023-07-26)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.23.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.1...@bigcommerce/big-design-icons@0.23.2) (2023-06-14)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.23.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.0...@bigcommerce/big-design-icons@0.23.1) (2023-06-09)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.23.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.22.0...@bigcommerce/big-design-icons@0.23.0) (2023-04-05)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.22.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.21.0...@bigcommerce/big-design-icons@0.22.0) (2023-01-09)

### Features

- convert to react@18 only ([#1044](https://github.com/bigcommerce/big-design/issues/1044)) ([06c4697](https://github.com/bigcommerce/big-design/commit/06c469721bc06d1f872bed6bf5a46e6b568644e5))

### BREAKING CHANGES

- Requires `react@18` or higher.

# [0.21.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.20.1...@bigcommerce/big-design-icons@0.21.0) (2022-09-27)

### Features

- **deps:** react@18 support ([51c688b](https://github.com/bigcommerce/big-design/commit/51c688b282277964eb01c1ed67c5c27cc0bc4c3e))

## [0.20.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.20.0...@bigcommerce/big-design-icons@0.20.1) (2022-09-27)

### Features

- **deps:** unfork from flag-icons dependency ([ac723e3](https://github.com/bigcommerce/big-design/commit/ac723e384ec3c482d78107558cd31e7793a515f1))

# [0.20.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.2-alpha.0...@bigcommerce/big-design-icons@0.20.0) (2022-08-17)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.19.2-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.1...@bigcommerce/big-design-icons@0.19.2-alpha.0) (2022-08-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.19.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.0...@bigcommerce/big-design-icons@0.19.1) (2022-07-12)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.19.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.18.1...@bigcommerce/big-design-icons@0.19.0) (2022-07-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.18.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.18.0...@bigcommerce/big-design-icons@0.18.1) (2022-06-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.18.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.2-alpha.1...@bigcommerce/big-design-icons@0.18.0) (2022-05-11)

### Bug Fixes

- tiny-async-pool api changes ([b4c04ba](https://github.com/bigcommerce/big-design/commit/b4c04bac3556b084ac1d39d2f0fde57efd7d8f43))

## [0.17.2-alpha.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.2-alpha.0...@bigcommerce/big-design-icons@0.17.2-alpha.1) (2022-03-22)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.17.2-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.1...@bigcommerce/big-design-icons@0.17.2-alpha.0) (2021-10-28)

### Bug Fixes

- **icons:** add `aria-hidden=true` if an icon doesn't have a title ([e3bb896](https://github.com/bigcommerce/big-design/commit/e3bb896fed2d7b1afb30f0134f161cfae5a00398))
- **icons:** Ensure aira-hidden is correctly set in flags ([9f49a2d](https://github.com/bigcommerce/big-design/commit/9f49a2dfd19734e797945483cb99efe8ea1503f1))

## [0.17.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0...@bigcommerce/big-design-icons@0.17.1) (2021-09-03)

### Features

- **icons:** Add Home/KeyboardDoubleArrowLeftIcon/SwapHoriz icons ([#583](https://github.com/bigcommerce/big-design/issues/583)) ([519655f](https://github.com/bigcommerce/big-design/commit/519655fe8ee3d619d1b85937e86635332fb77301))

# [0.17.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.2...@bigcommerce/big-design-icons@0.17.0) (2021-08-04)

### Features

- **icons:** add content copy icon ([#572](https://github.com/bigcommerce/big-design/issues/572)) ([4ca0b76](https://github.com/bigcommerce/big-design/commit/4ca0b76d12cc5a9d2e1dd0b90d90bc2cfb6a1294))

# [0.17.0-alpha.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.1...@bigcommerce/big-design-icons@0.17.0-alpha.2) (2021-08-02)

### chore

- **deps:** upgrade to styled-components v5 ([#566](https://github.com/bigcommerce/big-design/issues/566)) ([dd83711](https://github.com/bigcommerce/big-design/commit/dd83711797eb8aaa7a8406eebf1383116bff2420))

### BREAKING CHANGES

- **deps:** You will need to update to styled-components v5. In addition,
  you will need to import the base fonts in your <head> element.
  See the "Getting Started" page or README.md for an example.

# [0.17.0-alpha.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.0...@bigcommerce/big-design-icons@0.17.0-alpha.1) (2021-07-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.17.0-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.16.1...@bigcommerce/big-design-icons@0.17.0-alpha.0) (2021-06-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.16.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.16.0...@bigcommerce/big-design-icons@0.16.1) (2021-03-29)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.16.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.15.0...@bigcommerce/big-design-icons@0.16.0) (2021-03-08)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.15.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.14.1...@bigcommerce/big-design-icons@0.15.0) (2021-02-11)

### Bug Fixes

- **docs:** Specify styled-components version in the README ([#490](https://github.com/bigcommerce/big-design/issues/490)) ([c92403e](https://github.com/bigcommerce/big-design/commit/c92403edf8e29bf1afd3c9c163c37be6ffe17a6c))

### Features

- **icons:** add cloud upload icon ([#501](https://github.com/bigcommerce/big-design/issues/501)) ([aeb9aee](https://github.com/bigcommerce/big-design/commit/aeb9aeeb18a93064c61497fc439e7e82957c1a2e))
- **icons:** add flag icons ([#493](https://github.com/bigcommerce/big-design/issues/493)) ([0519c8a](https://github.com/bigcommerce/big-design/commit/0519c8aa131461e52ffe40f40b1e45771f749c0f))

## [0.14.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.14.0...@bigcommerce/big-design-icons@0.14.1) (2020-12-21)

### Features

- **icons:** add insert_drive_file icon ([#486](https://github.com/bigcommerce/big-design/issues/486)) ([b07d9c6](https://github.com/bigcommerce/big-design/commit/b07d9c6f6416854212eb447d216117a40edb6b8e))

# [0.14.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.13.1...@bigcommerce/big-design-icons@0.14.0) (2020-11-12)

### Bug Fixes

- webpack 5 compatibility ([#466](https://github.com/bigcommerce/big-design/issues/466)) ([7e66238](https://github.com/bigcommerce/big-design/commit/7e66238ca42cb27d91b1a80cc9e1c8014808e27b))

## [0.13.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.13.0...@bigcommerce/big-design-icons@0.13.1) (2020-08-31)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.13.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.12.0...@bigcommerce/big-design-icons@0.13.0) (2020-08-13)

### Features

- **icons:** add code, brush and get_app icons ([#433](https://github.com/bigcommerce/big-design/issues/433)) ([bb48f2d](https://github.com/bigcommerce/big-design/commit/bb48f2d6aefa627f35137f6043b582268ac76ef4))

# [0.12.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.11.1...@bigcommerce/big-design-icons@0.12.0) (2020-06-16)

### Features

- **component:** add Tree component ([#406](https://github.com/bigcommerce/big-design/issues/406)) ([c67643c](https://github.com/bigcommerce/big-design/commit/c67643cfdf425d078a1ea09bce12575e35f442e7))

## [0.11.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.11.0...@bigcommerce/big-design-icons@0.11.1) (2020-04-30)

### Bug Fixes

- type regression for ts < 3.8 ([#395](https://github.com/bigcommerce/big-design/issues/395)) ([f476f9b](https://github.com/bigcommerce/big-design/commit/f476f9b3c2a950bb3bd3353a4ce180e994e465c0))

# [0.11.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.10.0...@bigcommerce/big-design-icons@0.11.0) (2020-04-29)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.10.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.9.0...@bigcommerce/big-design-icons@0.10.0) (2020-04-20)

### Bug Fixes

- **icons:** contained aria-labelledby when title prop was undefined ([#374](https://github.com/bigcommerce/big-design/issues/374)) ([f0d4df7](https://github.com/bigcommerce/big-design/commit/f0d4df70894b364ec932dfc5c62cebda5dd10bda))

# [0.9.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.8.0...@bigcommerce/big-design-icons@0.9.0) (2020-03-25)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.8.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.7.0...@bigcommerce/big-design-icons@0.8.0) (2020-02-21)

### Features

- **component:** add Alerts, InlineAlerts, and Message components ([#340](https://github.com/bigcommerce/big-design/issues/340)) ([d541276](https://github.com/bigcommerce/big-design/commit/d54127603fba47b46cb35c3db4caf53ab24bafc3))

# [0.7.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.6.0...@bigcommerce/big-design-icons@0.7.0) (2020-02-06)

### Bug Fixes

- add react-hooks eslint and fix missing deps/issues ([#330](https://github.com/bigcommerce/big-design/issues/330)) ([da3fbd6](https://github.com/bigcommerce/big-design/commit/da3fbd68181e98e43a95de7fce9956be91afc9b8))

### Features

- **icons:** forwardRef to svg ([#312](https://github.com/bigcommerce/big-design/issues/312)) ([0088444](https://github.com/bigcommerce/big-design/commit/0088444015df4f6538d97e657f800f5718c2706e))

# [0.6.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.2...@bigcommerce/big-design-icons@0.6.0) (2019-12-09)

### Bug Fixes

- **icons:** only add title when available ([b2cf7f7](https://github.com/bigcommerce/big-design/commit/b2cf7f7a00a0d158c01485ee6978c2e13404c048))

### Features

- **component:** add tab trap for modals and dialogs ([dcacbf9](https://github.com/bigcommerce/big-design/commit/dcacbf96a38bef1134e2a8dcbd986f6362e0e2b7))
- **component:** ignore theme prop overrides ([24b92f9](https://github.com/bigcommerce/big-design/commit/24b92f9873fe4f51975b7bd00300c8ce73484ea3))

### BREAKING CHANGES

- **component:** `theme` prop override is no longer supported.

## [0.5.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.1...@bigcommerce/big-design-icons@0.5.2) (2019-11-14)

### Bug Fixes

- **icons:** show proper displayName for icons ([#268](https://github.com/bigcommerce/big-design/issues/268)) ([539a02a](https://github.com/bigcommerce/big-design/commit/539a02a))

## [0.5.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.0...@bigcommerce/big-design-icons@0.5.1) (2019-11-12)

### Features

- **all:** better tree shaking ([c0998a7](https://github.com/bigcommerce/big-design/commit/c0998a7))

# [0.5.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.4.0...@bigcommerce/big-design-icons@0.5.0) (2019-11-11)

### Features

- **all:** bump styled-components peerDependency version ([fd89fa3](https://github.com/bigcommerce/big-design/commit/fd89fa3))

### BREAKING CHANGES

- **all:** bumped peer dependency of styled-components to `^4.3.0`

# [0.4.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.3.0...@bigcommerce/big-design-icons@0.4.0) (2019-10-29)

### Features

- **icons:** add ArrowDownward and ArrowUpward icons ([c34203f](https://github.com/bigcommerce/big-design/commit/c34203f))

# [0.3.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.2.1...@bigcommerce/big-design-icons@0.3.0) (2019-09-23)

### Features

- **component:** add indeterminate state to checkboxes ([#197](https://github.com/bigcommerce/big-design/issues/197)) ([5146fdb](https://github.com/bigcommerce/big-design/commit/5146fdb))

## [0.2.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.2.0...@bigcommerce/big-design-icons@0.2.1) (2019-09-17)

### Bug Fixes

- **icons:** use new api on icon downloader ([#195](https://github.com/bigcommerce/big-design/issues/195)) ([6bae4ae](https://github.com/bigcommerce/big-design/commit/6bae4ae))

# [0.2.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.2...@bigcommerce/big-design-icons@0.2.0) (2019-08-20)

### Features

- **icons:** add BaselineHelpIcon ([#180](https://github.com/bigcommerce/big-design/issues/180)) ([e2387cf](https://github.com/bigcommerce/big-design/commit/e2387cf))

## [0.1.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.1...@bigcommerce/big-design-icons@0.1.2) (2019-08-19)

### Bug Fixes

- **icons:** now treeshakeable ([#178](https://github.com/bigcommerce/big-design/issues/178)) ([fb3773d](https://github.com/bigcommerce/big-design/commit/fb3773d))

## [0.1.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.0...@bigcommerce/big-design-icons@0.1.1) (2019-08-15)

### Features

- **docs:** add responsive mobile header ([#165](https://github.com/bigcommerce/big-design/issues/165)) ([dca031a](https://github.com/bigcommerce/big-design/commit/dca031a))

# 0.1.0 (2019-08-13)

### Bug Fixes

- **icons:** point to the correct type definitions ([f6ffb48](https://github.com/bigcommerce/big-design/commit/f6ffb48))
- **icons:** star icon now displays a star and not a square ([331e71b](https://github.com/bigcommerce/big-design/commit/331e71b))
- **icons:** use object style notation for icon base ([41a2638](https://github.com/bigcommerce/big-design/commit/41a2638))
- **theme:** issue with font sizes in createTheme ([#135](https://github.com/bigcommerce/big-design/issues/135)) ([313ff46](https://github.com/bigcommerce/big-design/commit/313ff46))

### Features

- **icons:** add color and size props ([ba07b38](https://github.com/bigcommerce/big-design/commit/ba07b38))
- **icons:** add initial batch of icons ([243d1a2](https://github.com/bigcommerce/big-design/commit/243d1a2))
- **icons:** add Notifications and FileCopy icons ([#103](https://github.com/bigcommerce/big-design/issues/103)) ([e385b9b](https://github.com/bigcommerce/big-design/commit/e385b9b))
- **icons:** initial package setup ([#89](https://github.com/bigcommerce/big-design/issues/89)) ([acfe8fe](https://github.com/bigcommerce/big-design/commit/acfe8fe))
- **icons:** use Icon postfix ([#97](https://github.com/bigcommerce/big-design/issues/97)) ([03772fb](https://github.com/bigcommerce/big-design/commit/03772fb))
