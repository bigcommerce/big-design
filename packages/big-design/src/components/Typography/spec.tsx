import { theme } from '@bigcommerce/big-design-theme';
import { render } from '@testing-library/react';
import React from 'react';

import 'jest-styled-components';

import { H0, H1, H2, H3, H4, HR, Small, Text } from './Typography';

test('render H0', () => {
  const { container } = render(<H0>Test</H0>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H1', () => {
  const { container } = render(<H1>Test</H1>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H2', () => {
  const { container } = render(<H2>Test</H2>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H3', () => {
  const { container } = render(<H3>Test</H3>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render H4', () => {
  const { container } = render(<H4>Test</H4>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Small', () => {
  const { container } = render(<Small>Test</Small>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Text', () => {
  const { container } = render(<Text>Test</Text>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render HR', () => {
  const { container } = render(<HR />);

  expect(container.firstChild).toMatchSnapshot();
});

test('render Text with ellipsis', () => {
  const { container } = render(<Text ellipsis={true}>Test with ellipsis</Text>);

  expect(container.firstChild).toMatchSnapshot();
});

test('H0 - does not forward styles', () => {
  const { container } = render(
    <H0 className="test" style={{ background: 'red' }}>
      Test
    </H0>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H1 - does not forward styles', () => {
  const { container } = render(
    <H1 className="test" style={{ background: 'red' }}>
      Test
    </H1>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H2 - does not forward styles', () => {
  const { container } = render(
    <H2 className="test" style={{ background: 'red' }}>
      Test
    </H2>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H3 - does not forward styles', () => {
  const { container } = render(
    <H3 className="test" style={{ background: 'red' }}>
      Test
    </H3>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('H4 - does not forward styles', () => {
  const { container } = render(
    <H4 className="test" style={{ background: 'red' }}>
      Test
    </H4>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('Small - does not forward styles', () => {
  const { container } = render(
    <Small className="test" style={{ background: 'red' }}>
      Test
    </Small>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('Text - does not forward styles', () => {
  const { container } = render(
    <Text className="test" style={{ background: 'red' }}>
      Test
    </Text>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('HR - does not forward styles', () => {
  const { container } = render(<HR className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test')).toHaveLength(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('All typography components allow changing their color given a color prop', () => {
  const { container } = render(
    <>
      <H1 color="primary">Test</H1>
      <H2 color="primary">Test</H2>
      <H3 color="primary">Test</H3>
      <H4 color="primary">Test</H4>
      <Small color="primary">Test</Small>
      <Text color="primary">Test</Text>
    </>,
  );

  Array.from(container.children).forEach((child) => {
    expect(child).toHaveStyle(`color: ${theme.colors.primary}`);
  });
});

test('HR allows changing its color given a color prop', () => {
  const { container } = render(<HR color="primary" />);

  expect(container.firstChild).toHaveStyle(`border-bottom: 1px solid ${theme.colors.primary}`);
});

test('HR can change its margins given a margin prop', () => {
  const { container } = render(<HR marginTop="medium" />);

  expect(container.firstChild).toHaveStyle(`margin-top: ${theme.spacing.medium}`);
});

test('Headings can change their tag', () => {
  const { container } = render(<H2 as="h1">Test</H2>);

  expect(container.querySelector('h1')).toBeInTheDocument();
});

test('Headings can not change their tag to non-heading tags', () => {
  // @ts-expect-error ignoring since as="p" is not a valid prop
  const { container } = render(<H2 as="p">Test</H2>);

  expect(container.querySelector('p')).not.toBeInTheDocument();
  expect(container.querySelector('h2')).toBeInTheDocument();
});

test('Text and Small can change their tag', () => {
  const { container } = render(
    <>
      <Text as="span">Some Text</Text>
      <Small as="span">Some Text</Small>
    </>,
  );

  expect(container.querySelectorAll('span')).toHaveLength(2);
});

test('Text and Small accept text modifiers', () => {
  const { getByTestId } = render(
    <>
      <Text bold data-testid="text" italic underline uppercase>
        Some Text
      </Text>
      <Small data-testid="small" lowercase strikethrough>
        Some Text
      </Small>
    </>,
  );

  const text = getByTestId('text');
  const small = getByTestId('small');

  expect(text).toHaveStyle(`
    font-weight: 600;
    font-style: italic;
    text-decoration: underline;
    text-transform: uppercase;
  `);

  expect(small).toHaveStyle(`
    text-decoration: line-through;
    text-transform: lowercase;
  `);
});
