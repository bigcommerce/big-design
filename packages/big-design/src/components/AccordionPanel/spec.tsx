import { ErrorIcon } from '@bigcommerce/big-design-icons';
import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import React, { useState } from 'react';

import { Text } from '../Typography';

import { AccordionProps } from './Accordion';

import { AccordionPanel, useAccordionPanel } from '.';

const TestComponent: React.FC = () => {
  const { panels } = useAccordionPanel([
    {
      header: 'Panel Header',
      children: <Text>This is a child component</Text>,
    },
  ]);

  return <AccordionPanel header="Accordion Panel Header" panels={panels} />;
};

const TestComponentWithoutHook: React.FC = () => {
  const [isExpanded, setIsExpanded] = useState(false);

  const testPanels: AccordionProps[] = [
    {
      header: 'Panel Header',
      isExpanded,
      children: <Text>This is a child component</Text>,
      onClick: () => setIsExpanded(!isExpanded),
    },
  ];

  return <AccordionPanel header="Accordion Panel Header" panels={testPanels} />;
};

const TestComponentWithDefault: React.FC = () => {
  const { panels } = useAccordionPanel([
    {
      defaultExpanded: true,
      header: 'Panel Header',
      iconLeft: <ErrorIcon />,
      children: <Text>This is a child component</Text>,
    },
  ]);

  return <AccordionPanel header="Accordion Panel Header" panels={panels} />;
};

test('it renders accordion panel header', async () => {
  render(<TestComponent />);

  const header = await screen.findByRole('heading', { name: /Accordion Panel Header/i });

  expect(header).toMatchSnapshot();
  expect(header).toBeVisible();
});

test('it renders accordion panel children', async () => {
  render(<TestComponent />);

  const child = await screen.findByText('Panel Header');

  expect(child).toBeVisible();
});

test('accordion renders header', async () => {
  render(<TestComponent />);

  const header = await screen.findByText('Panel Header');

  expect(header).toBeVisible();
});

test('accordion collapses and expands on click', async () => {
  render(<TestComponent />);

  const buttonPanel = await screen.findByRole('button');

  await userEvent.click(buttonPanel);

  const panelChild = await screen.findByRole('region');

  expect(panelChild).toBeVisible();

  await userEvent.click(buttonPanel);

  expect(panelChild).not.toBeVisible();
});

test('accordion renders children', async () => {
  render(<TestComponent />);

  const buttonPanel = await screen.findByRole('button');

  await userEvent.click(buttonPanel);

  const child = await screen.findByRole('region');

  expect(child).toBeVisible();
});

test('if defaultExpanded is set to true, accordion panel is expanded', async () => {
  render(<TestComponentWithDefault />);

  const panelChild = await screen.findByRole('region');

  expect(panelChild).toBeVisible();
});

test('it renders icon if iconLeft prop is defined', async () => {
  render(<TestComponentWithDefault />);

  const button = await screen.findByRole('button');

  // eslint-disable-next-line testing-library/no-node-access
  expect(button.querySelector('svg')).toBeInTheDocument();
});

test('it renders header and children accordion elements without hook', async () => {
  render(<TestComponentWithoutHook />);

  const header = await screen.findByText('Panel Header');
  const buttonPanel = await screen.findByRole('button');

  await userEvent.click(buttonPanel);

  const child = await screen.findByRole('region');

  expect(child).toBeVisible();
  expect(header).toBeVisible();
});
