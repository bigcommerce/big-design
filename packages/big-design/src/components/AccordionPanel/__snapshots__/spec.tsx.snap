// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`it renders accordion panel header 1`] = `
<h2
  class="styled__StyledH2-sc-218b28a9-2 styled__StyledH2-sc-eda6cb79-1 hCxYWp cMjtfK"
>
  Accordion Panel Header
</h2>
`;
